code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -120,7 +120,15 @@
 	if(!isliving(M))
 		return
 
-	var/mob/living/L = M
+	var/mob/living/carbon/human/L = M
+
+	if(L.martial_art && L.martial_art.block_chance)

```

Comment:
because batons dont trigger the attacked proc in species so now all procs have to go to the batons attack proc itself, this is the same for checking for shields and humanity, it all happens in the stunbaton's proc.

misc batons are in melee misc